src/contract/ContractCreateFlow.js

```diff
@@ -254,4 +258,80 @@ export default class ContractCreateFlow {
 
         return response;
     }
+
+    /**
+     * @param {Signer} signer
+     * @returns {Promise<TransactionResponse>}
+     */
+    async executeWithSigner(signer) {
+        if (this._bytecode == null) {
+            throw new Error("cannot create contract with no bytecode");
+        }
+
+        if (signer.getAccountKey == null) {
+            throw new Error(
+                "`Signer.getAccountKey()` is not implemented, but is requried for `ContractCreateFlow`"
+            );
+        }
+
+        const key = signer.getAccountKey();
+
+        let response = await new FileCreateTransaction()
+            .setKeys(key != null ? [key] : [])
+            .setContents(
+                this._bytecode.subarray(
+                    0,
+                    Math.min(this._bytecode.length, 2048)
+                )
+            )
+            .executeWithSigner(signer);
+
+        const receipt = await getReceiptWithSigner(response, signer);
+
+        const fileId = /** @type {FileId} */ (receipt.fileId);
+
+        if (this._bytecode.length > 2048) {
+            await new FileAppendTransaction()
+                .setFileId(fileId)
+                .setContents(this._bytecode.subarray(2048))
+                .executeWithSigner(signer);
+        }
+
+        response = await this._contractCreate
+            .setBytecodeFileId(fileId)
+            .executeWithSigner(signer);
+
+        await getReceiptWithSigner(response, signer);
+
+        if (key != null) {
+            const fileDeleteResponse = await new FileDeleteTransaction()
+                .setFileId(fileId)
+                .executeWithSigner(signer);
+
+            await getReceiptWithSigner(fileDeleteResponse, signer);
+        }
+
+        return response;
+    }
+}
+
+/**
+ * Perhaps we should implement this on `TransactionResponse`?
```

Comment:
I definitely agree with this. Since `Signer` takes the place of `Client` in the `*WithSigner()` methods, implementing `get[Receipt|Record]WithSigner()` on `TransactionResponse` makes perfect since for any transactions/flows/etc using the former. Sounds like a really convenient way to not have to involve `Client` in places where it otherwise isn't used.